enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,598 @@
+
+When generating the BundleDeployment resource, the PO manager will use the unique "core.rukpak.io/plain" provisioner ID to manage and apply the desired Bundle template, and the "core.rukpak.io/registry" provisioner ID to source the desired registry+v1 bundle contents.
+
+An example of this behavior can be seen below:
+
+```yaml=
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleDeployment
+metadata:
+  name: openshift-cert-manager.v1.7.1
+spec:
+  provisionerClassName: core.rukpak.io/plain
+  template:
+    spec:
+      provisionerClassName: core.rukpak.io/registry
+      source:
+        type: image
+        image:
+          ref: registry.redhat.io/cert-manager/cert-manager-operator-bundle:v4.12.0
+```
+
+After applying the BundleDeployment resource to the cluster, the PO manager delegates the lifecycle of this bundle to the plain and registry OLM provisioners.
+
+OLM's existing plain provisioner is responsible for managing any BundleDeployment resources that specify the "core.rukpak.io/plain" unique provisioner ID. This provisioner knows how to interact with plain+v0 ("plain") bundle formats. A plain bundle is simply a collection of static, arbitrary, Kubernetes YAML manifests in a given directory.
+
+When the plain provisioner's BundleDeployment controller sees a BundleDeployment resource that contains a desired Bundle template, and that template doesn't match any Bundles that exist on the cluster, a new Bundle will be generated to match that desired configuration.
+In this case, the plain provisioner will generate a Bundle that gets sourced by the registry provisioner, and wait until that generated Bundle reports an unpacked phase.
+
+The registry provisioner has logic that converts OLM’s registry+v1 bundle format into a plain bundle format. The OLM ClusterServiceVersion (CSV) resource can be decomposed into a set of core Kubernetes manifests (e.g. RBAC, Deployment, etc.).
+This conversion logic allows the registry provisioner to effectively act as a bridge between higher-level components (e.g. PO manager) that want to apply registry+v1 bundle contents, and still have the rukpak stack manage those contents.
+
+After the registry provisioner successfully unpacks the registry+v1 bundle, and converts that into a plain bundle format, the plain provisioner will attempt to apply those manifests. In the case that an individual manifest is invalid, and the Kubernetes API server rejects the creation of that manifest, then the plain provisioner will update the BundleDeployment resource to reflect that failed rollout.
+
+### Workflow Description
+
+```mermaid
+sequenceDiagram
+actor Admin
+participant Installer
+participant Bootstrap
+participant CVO
+participant ClusterOperator
+participant PO manager
+Installer->>Admin: Creates the Kubernetes manifests
+CVO->>PO custom resource: Generates
```

Comment:
CVO applies the PO custom resource (supplied by the admin), right?  not generates.

Reply:
Updated in https://github.com/openshift/enhancements/pull/1170/commits/e6f356ef53203a97ab98f654704310f1e8a2f790.